src/integer_mod_q/poly_over_zq/get.rs

```diff
@@ -94,6 +94,24 @@ impl GetCoefficient<Z> for PolyOverZq {
     }
 }
 
+impl PolyOverZq {
+    /// Returns the degree of a polynomial [`PolyOverZq`] as a [`i64`].
+    /// The zero polynomial has degree '-1'.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::PolyOverZq;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZq::from_str("4  0 1 2 3 mod 7").unwrap();
+    ///
+    /// let degree = poly.get_degree(); // This would only return 3
+    /// ```
+    pub fn get_degree(&self) -> i64 {
+        unsafe { fmpz_mod_poly_degree(&self.poly, &*self.modulus.modulus) }
```

Comment:
@MoogSven does this lead to a memory leak?

Reply:
There is no memory leak. However, it could be simplified to:
```rs
self.poly.length - 1
```
This is also what happens in FLINT.